xdsl/dialects/gpu.py

```diff
@@ -251,6 +251,26 @@ def get(dim: DimensionAttr) -> BlockIdOp:
                                attributes={"dimension": dim})
 
 
+@irdl_op_definition
+class DeallocOp(Operation):
+    name = "gpu.dealloc"
+
+    asyncDependencies: Annotated[VarOperand, AsyncTokenType]
+    buffer: Annotated[Operand, memref.MemRefType]
+
+    irdl_options = [AttrSizedOperandSegments()]
+
+    asyncToken: Annotated[OptOpResult, AsyncTokenType]
+
+    @staticmethod
+    def get(buffer: SSAValue | Operation,
+            async_dependencies: Sequence[SSAValue | Operation] | None = None,
+            is_async: bool = False) -> DeallocOp:
+        return DeallocOp.build(
+            operands=[async_dependencies or [], buffer],
```

Comment:
Wait, how does this work? Or on lists? :sweat_smile:

Reply:
I thought about it following your comment on the other PR! it works this way basically: 
```pycon
>>> a = 3
>>> b = None
>>> a or 0
3
>>> b or 0
0
```

Reply:
So if one side is None, it just gives the other? I guess I just haven't seen that often enough yet. I am find with leaving as is :)

Reply:
Sounds like a bugcatching hell if it goes wrong in some constructor at some point..

Reply:
Yeah, agreed. Isn't Python bug-catching hell anyways though :P

Reply:
Thinking about it: The `async_depencies if async_dependencies else []` is exactly the same thing...
Now for list, it's pretty straightforward, because it would be false for `None` or `[]`
I'm sticking to `is not None` for now, but that sounds like something worth deciding somewhere then?